ipa-core/src/net/http_serde.rs

```diff
@@ -533,4 +533,78 @@ pub mod query {
 
         pub const AXUM_PATH: &str = "/:query_id/complete";
     }
+
+    pub mod kill {
+        use serde::{Deserialize, Serialize};
+
+        use crate::{
+            helpers::{routing::RouteId, HelperResponse, NoStep, RouteParams},
+            protocol::QueryId,
+        };
+
+        #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+        pub struct Request {
+            pub query_id: QueryId,
+        }
+
+        impl RouteParams<RouteId, QueryId, NoStep> for Request {
+            type Params = String;
+
+            fn resource_identifier(&self) -> RouteId {
+                RouteId::KillQuery
+            }
+
+            fn query_id(&self) -> QueryId {
+                self.query_id
+            }
+
+            fn gate(&self) -> NoStep {
+                NoStep
+            }
+
+            fn extra(&self) -> Self::Params {
+                String::new()
+            }
+        }
+
+        impl Request {
+            /// see above for the reason why this needs to be behind a feature flag
```

Comment:
why are these different feature flags? does `cli` need to be included, so that this is available when run from the cli as a helper?

Reply:
yea, I changed the comment because it didn't really explain why those specific features were used to gate